### PR TITLE
Issue #477: Abstract Qbeast Catalog Tests from Delta

### PR DESCRIPTION
**Pull request by JosepSampe**
## Description

Abstracts catalog tests from delta #477 
Included schema in the `IndexedTable` for flexibility

## Type of change

Refactoring (#398)

## Checklist:

Here is the list of things you should do before submitting this pull request:

- [x] New feature / bug fix has been committed following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [x] Add logging to the code following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [x] Add comments to the code (make it easier for the community!).
- [x] Change the documentation.
- [x] Add tests.
- [x] Your branch is updated to the main branch (dependent changes have been merged).

### CODE REVIEW

File: src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala [hunk 1]
```diff
@@ -138,15 +128,11 @@ object QbeastCatalogUtils extends Logging {
...
-          throw AnalysisExceptionFactory
-            .create(
-              "Trying to create a managed table with a different schema. " +
-                "Do you want to want to ALTER TABLE first?")
-        }
+      if (indexedTableExists && indexedTable.schema != table.schema) {
+        throw AnalysisExceptionFactory
+          .create(
+            "Trying to create a managed table with a different schema. " +
+              "Do you want to want to ALTER TABLE first?")
```
Comment (osopardo1) [inline src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala:135]: Do you want to... (is repeated twice)

File: src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala [hunk 2]
```diff
@@ -110,21 +106,15 @@ object QbeastCatalogUtils extends Logging {
...
 
-    val isTablePopulated = table.tableType == CatalogTableType.EXTERNAL && fs
-      .exists(path) && fs.listStatus(path).nonEmpty
-    // Users did not specify the schema. We expect the schema exists in Delta.
     if (table.schema.isEmpty) {
+      // Users did not specify the schema. We expect the schema exists
       if (table.tableType == CatalogTableType.EXTERNAL) {
-        if (fs.exists(path) && fs.listStatus(path).nonEmpty) {
-          val existingSchema =
-            DeltaLog.forTable(spark, path.toString).unsafeVolatileSnapshot.metadata.schema
```
Comment (osopardo1) [inline src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala:126]: oh that was a good catch.. why didn't we detected the DeltaLog import before?

File: src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala [hunk 3]
```diff
@@ -110,21 +106,15 @@ object QbeastCatalogUtils extends Logging {
     }
   }
 
-  private def verifySchema(
-      spark: SparkSession,
-      fs: FileSystem,
-      path: Path,
-      table: CatalogTable): CatalogTable = {
+  private def verifySchema(indexedTable: IndexedTable, table: CatalogTable): CatalogTable = {
```
Comment (osopardo1) [inline src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala:109]: We should add some comments about what is the scope of this method (I know it should've been written before, sorry for that :((( )

File: src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala (hunk 1 above)
Comment (JosepSampe) [reply src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala]: FIxed

File: src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala (hunk 3 above)
Comment (JosepSampe) [reply src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala:124]: Added

File: src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala (hunk 2 above)
Comment (JosepSampe) [reply src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala:126]: Good question, the file ended in the src/ folder were we have Delta, so it didn't raise any exception

**Review by osopardo1 — APPROVED** [review]

**Pull request merged**